R/CallbackSetUnfreeze.R

```diff
@@ -0,0 +1,73 @@
+#' @title Unfreezing Weights Callback
+#'
+#' @name mlr_callback_set.unfreeze
+#'
+#' @description
+#' Unfreeze some weights after some number of steps or epochs. Select either a given module or a parameter.
+#'
+#' @details
+#' TODO: add
+#'
+#' @param starting_weights (`Select`)\cr
+#'  A `Select` denoting the weights that are trainable from the start.
+#' @param unfreeze (`data.table`)\cr
+#'  A `data.table` with a column `weights` (a list column containing a `Select`) and a column `epoch` or `batch`.
+#'
+#' @family Callback
+#' @export
+#' @include CallbackSet.R
+CallbackSetUnfreeze = R6Class("CallbackSetUnfreeze",
+  inherit = CallbackSet,
+  lock_objects = FALSE,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(starting_weights, unfreeze) {
+      self$starting_weights = starting_weights
+      # consider supporting character vectors
+      self$unfreeze = unfreeze
+    },
+    #' @description
+    #' Sets the starting weights
+    on_begin = function() {
+      weights = select_invert(self$starting_weights)(names(self$ctx$network$parameters))
```

Comment:
maybe we should also make sure to call `requires_grad_(TRUE)` on those weights that we want to train (just to be on the safe side)